transport/grpc/dial.go

```diff
@@ -184,9 +184,37 @@ func dial(ctx context.Context, insecure bool, o *internal.DialSettings) (*grpc.C
 		grpcOpts = append(grpcOpts, timeoutDialerOption)
 	}
 
+	// TODO(cbro): support "autodetect" and "never", and possibly merge with/refactor the equivalent in htransport.
+	if os.Getenv("GOOGLE_API_USE_MTLS") == "always" {
```

Comment:
Is this different from `GOOGLE_API_USE_CLIENT_CERTIFICATE` discussed in ADC design doc?

Reply:
Also, should this apply to HTTP as well?

Reply:
Already implemented in HTTP. This CL is giving the minimum support for the feature for our nightly smoke test (critical - can give more background over chat).

But yes, the semantics and name changed, so there'll be a separate implementation for GOOGLE_API_USE_CLIENT_CERTIFICATE in the future.

GOOGLE_API_USE_TLS will become only for the smoketest. I'll update the comment to reflect that.

Reply:
Thanks for the clarification.